### PR TITLE
[BugFix][Model] Support DeepSeek-OCR2 on 310P

### LINKED ISSUE #10667: [Contribution] 任务 #20：310P DeepSeek-OCR2 多模态OCR模型适配优化
**在提交新问题之前，请确保您已经在社区中搜索过相关问题，并使用了社区中提供的资源/工具后，仍未找到满意的解决方式。**

## 任务描述
该任务拆分自 #9079，任务 ID：#20。

适配DeepSeek-OCR2高精度图文识别模型，该模型支持多语言文字识别、手写体识别、复杂版式文档解析；基于vLLM Ascend框架完成图像预处理、视觉编码、文字解码链路适配，兼容Ascend NPU算力特性，优化图像推理算力占用，保障OCR识别精度与推理速度达标。

## 验收标准
1. 在 `tests/e2e/models/configs` 添加完整测试用例。
2. 更新模型支持清单文档，编写OCR模型专属适配部署教程。
3. 提交 `SKILL.md`，记录多模态模型适配技巧、算子优化方案及开发最佳实践。

## 对接人
@jyoung6652

## 认领状态
待认领

### 建议的替代/修复方案
请根据任务描述提交实现、测试或文档 PR，并在 PR 中关联本 issue 与 #9079；如实现过程中发现任务边界、使用指导或现有行为存在不合理之处，请同步补充说明和最小复现。

欢迎加入社区，感谢您对社区的贡献 🎉!

### PR DESCRIPTION
**Pull request by treason258**
### What this PR does / why we need it?

Fixes #10667. Ref #9079.

This PR adapts DeepSeek-OCR-2 for Atlas 300I DUO (310P) single-card serving:

- Adds 310P-specific backend fixes for DeepSeek-OCR2 FP16 paths, including ND linear/LM head handling, rel-pos attention fallback, RoPE edge cases, tokenizer backend selection, and Qwen2 decoder compatibility.
- Adds the DeepSeek-OCR2 310P e2e model config.
- Updates the supported-model matrix and DeepSeekOCR2 tutorial with the verified 310P deployment command, functional check, TextVQA result, and single-card batch sweep.
- Updates the existing model-adapter skill reference with concise 310P multimodal adaptation lessons.

Known scope: the signed-off 310P path is FP16. 310P does not support BF16; INT8/INT4 should be validated as separate quantized paths. OmniDocBench v1.5 result is OverallEdit `10.8706` (paper: `10.0`), tracked as a follow-up.

### Does this PR introduce _any_ user-facing change?

Yes. DeepSeek-OCR-2 can be served on a single 310P card with the documented FP16 command, and the model support documentation now lists 310P.

### How was this patch tested?

- `python -m py_compile vllm_ascend/__init__.py vllm_ascend/utils.py vllm_ascend/ops/linear.py vllm_ascend/ops/qwen2_decoder.py vllm_ascend/ops/rel_pos_attention.py vllm_ascend/_310p/ops/rotary_embedding.py vllm_ascend/_310p/ops/vocab_parallel_embedding.py`
- `pytest -q tests/ut/test_utils.py::TestUtils::test_deepseek_ocr2_tokenizer_backend_patch tests/ut/_310p/ops/test_vocab_parallel_embedding.py`
- `python .github/workflows/scripts/coverage.py`
- `uvx pre-commit run ruff-format --all-files && uvx pre-commit run ruff-check --all-files`
- `python /root/.codex/skills/.system/skill-creator/scripts/quick_validate.py .agents/skills/vllm-ascend-model-adapter`
- `npx --yes markdownlint-cli2 docs/source/tutorials/models/DeepSeekOCR2.md .agents/skills/vllm-ascend-model-adapter/references/multimodal-ep-aclgraph-lessons.md`
- `git diff --check`

Local real-checkpoint 310P validation performed:

- `/v1/models`, Free OCR image request, grounding image request, and repeated image requests returned HTTP 200.
- ACLGraph replay was enabled and logs showed `Replaying aclgraph`.
- Single-card `max-num-seqs=4` mixed OCR/grounding soak completed 80/80 HTTP 200 requests, followed by a 1024-token long-output request and post-long short requests.
- TextVQA-lite result: `accuracy,gen = 0.4982`.

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/b9a7cd464c9ae9b1b450f8982b76d7be4de73724

### CODE REVIEW

File: vllm_ascend/ops/rel_pos_attention.py [hunk 1]
```diff
@@ -2,6 +2,108 @@
...
+    v = v.reshape(batch_size * num_heads, v.size(-2), head_dim)
+    attn = torch.bmm(q, k) * scale
+    attn = torch.softmax(attn, dim=-1)
+    attn = attn.view(batch_size * num_heads, query_tokens, k_h, k_w)
+    rel_h = rel_h.reshape(batch_size * num_heads, query_tokens, k_h, 1)
+    rel_w = rel_w.reshape(batch_size * num_heads, query_tokens, 1, k_w)
+    attn = attn * torch.exp(rel_h)
+    attn = attn * torch.exp(rel_w)
+    attn = attn.reshape(batch_size * num_heads, query_tokens, k_h * k_w)
+    attn = torch.softmax(torch.log(attn), dim=-1)
```
[the remark this answers is not in the document]
Comment (treason258) [reply vllm_ascend/ops/rel_pos_attention.py]: Addressed in e0ab0ef: `_rel_pos_attention_310p` now adds relative-position biases to the logits and applies a single softmax, avoiding the previous softmax/exp/log/softmax sequence.

File: vllm_ascend/_310p/ops/vocab_parallel_embedding.py [hunk 2]
```diff
@@ -24,19 +24,37 @@
...
         self,
         layer: torch.nn.Module,
         x: torch.Tensor,
         bias: torch.Tensor | None = None,
     ) -> torch.Tensor:
+        if getattr(layer, "_deepseek_ocr2_310p_nd_lm_head", False):
+            if x.dtype == torch.bfloat16 and layer.weight.dtype == torch.float16:
+                x = x.to(torch.float16)
+            logits = torch.matmul(x, layer.weight.t())
+            return logits if bias is None else logits + bias
```
[the remark this answers is not in the document]
Comment (treason258) [reply vllm_ascend/_310p/ops/vocab_parallel_embedding.py]: Addressed in e0ab0ef: the 310P DeepSeek-OCR2 LM-head path now preserves the input dtype after the matmul, casts bias to the matmul dtype, and has a unit test for the BF16-input/FP16-weight case.
